Source/CombatExtended/CombatExtended/Verbs/Verb_LaunchProjectileCE.cs

```diff
@@ -348,9 +348,157 @@ public virtual void ShiftTarget(ShiftVecReport report, bool calculateMechanicalO
                     }
                     else if (currentTarget.Thing is Pawn)
                     {
-                        // Aim for center of mass on an exposed target
-                        targetRange.min = victimVert.BottomHeight;
+                        targetRange.min = victimVert.BottomHeight;
                         targetRange.max = victimVert.MiddleHeight;
+
+                        bool AppropiateAimMode = CompFireModes?.CurrentAimMode != AimMode.SuppressFire;
+
+                        bool flagautomatic = (CompFireModes?.target_mode != TargettingMode.automatic);
+
+                        if ( (CasterPawn?.Faction ?? Caster.Faction) == Faction.OfPlayer && flagautomatic)
+                        {
+                            bool flag1 = ((ShootingAccuracy >= 2.2f) | (Caster is Building_TurretGunCE));

```

Comment:
You set this `bool` and use it once, immediately after.  And the next block contains a single if.

```C#
if (AppropriateAimMode && (ShootingAccuracy >= 2.2f | Caster is Building_TurretGunCE))
```

If there is a way to *know* Handling has been read already, the last clause gets replaced with `IsTurret`, or `IsTurret` could become a getter itself, or be included in the ShootingAccuracy getter.

Reply:
This change will trigger the else block, where ShootingAccuracy is checked again.
Probably better to hoist the ShootingAccuracy check to the top, and then check aim mode.